dali/benchmark/transpose_bench.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021, Konrad Litwiński. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <benchmark/benchmark.h>
+#include "dali/benchmark/operator_bench.h"
+#include "dali/benchmark/dali_bench.h"
+
+namespace dali {
+
+static void TransposeGPUArgs(benchmark::internal::Benchmark *b) {
+  for (int batch_size = 256; batch_size >= 1; batch_size /= 2) {
+    for (int H = 2000; H >= 500; H /= 2) {
+      int W = H;
+      b->Args({batch_size, H, W});
+    }
+  }
+}
+
+BENCHMARK_DEFINE_F(OperatorBench, TransposeGPU)(benchmark::State& st) {
+  int batch_size = st.range(0);
+  int H = st.range(1);
+  int W = st.range(2);
+  std::vector<int> perm = {2, 0, 1};
```

Comment:
Please add more permutations and dimensions. The performance varies *vastly* depending on what permutation is done. Also, the sizes do matter for performance. To put that in perspective:
transposing 10000x3 to 3x10000 uses a different kernel than transposing 300x100 to 100x300 depsite using the same permutation and having the same total size.
Note that there is some simplification being done in Transpose - for example:
when you have a tensor of size AxBxCxD and you transpose it with (2, 3, 0, 1) it will be simplified into transposing
a (A*B)x(C*D) tensor with (1, 0) permutation.

The actual algorithm is chosen after simplification is applied and it depends on the extent of the innermost non-transposed dimension and the innermost transposed dimension - for example:
300x400x3 -> 400x300x3 and 300x400x32 -> 400x300x32 will likely use different kernels
likewise, 300x400x3 -> 400x300x3 and 300x32x3 -> 32x300x3 will also use different kernels, AFAIR.